apps/api/src/app/events/usecases/process-subscriber/process-subscriber.usecase.ts

```diff
@@ -113,14 +102,30 @@ export class ProcessSubscriber {
         _templateId: notification._templateId,
         digest: step.metadata,
         type: step.template.type,
-        providerId: integration?.providerId ?? InAppProviderIdEnum.Novu,
+        providerId: providerId,
         ...(actorSubscriber && { _actorId: actorSubscriber._id }),
       });
     }
 
     return jobs;
   }
 
+  private async getProviderId(command: ProcessSubscriberCommand, stepType: StepTypeEnum): Promise<string | null> {
+    const channelType = ChannelTypeEnum[stepType.toUpperCase()];
```

Comment:
I feel like it's better to create a map object instead of doing `.toUpperCase()`, it will be more type-safe

Reply:
Could you explain which map object we need to create? you mean convert ChannelTypeEnum to Map?